### DISCUSSION TITLE
Select solver when using Clp via OSI

### DISCUSSION THREAD

**Discussion by InaL9141** [coin-or/Clp#254] [Q&A] [answered]
I want to solve linear programs using Clp via the `OsiClpSolverInterface`. Using  `OsiClpSolverInterface`s `initialSolve` and `resolve` methods, I can solve the linear programs. Which of the Clp solvers is called in this case (primal or dual simplex, interior point)? Is it possible to select the Clp solver using the interface?
My motivation is that I currently get a lot of degenerate solutions. I want to try out whether this behaviour persists when using a different solver.

Thank you in advance for your help!
Ina

**Comment by jjhforrest** [comment, marked as answer]
To choose between dual and primal, you can use a "hint" - this is true for all Osi solvers
        solver->setHintParam(OsiDoDualInResolve, false, OsiHintTry);

To have more flexibility you would have to go via solver->getModelPtr() which is ClpSimplex*.  This would also allow you to use barrier.
